extension/httpforwarder/extension.go

```diff
@@ -16,21 +16,101 @@ package httpforwarder
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
 
 	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
 )
 
 type httpForwarder struct {
+	forwardTo  *url.URL
+	httpClient *http.Client
+	server     *http.Server
+	logger     *zap.Logger
+	config     *Config
 }
 
 var _ component.ServiceExtension = (*httpForwarder)(nil)
 
-func (h httpForwarder) Start(_ context.Context, _ component.Host) error {
-	// TODO: Start HTTP server
+func (h *httpForwarder) Start(_ context.Context, host component.Host) error {
+	listener, err := h.config.Ingress.ToListener()
+	if err != nil {
+		return fmt.Errorf("failed to bind to address %s: %w", h.config.Egress.Endpoint, err)
```

Comment:
Wouldn't we want to log`h.config.Ingress.Endpoint` here?

Reply:
Fixed.